test/e2e/dns.go

```diff
@@ -0,0 +1,409 @@
+		oc, err := clients.OpenshiftClusters.Get(ctx, vnetResourceGroup, clusterName)
+		Expect(err).NotTo(HaveOccurred())
+
+		clusterResourceGroup := stringutils.LastTokenByte(*oc.OpenShiftClusterProperties.ClusterProfile.ResourceGroupID, '/')
+		for wn := range workerNodes {
+			nic, err := clients.Interfaces.Get(ctx, clusterResourceGroup, nicName(wn), "")
+			Expect(err).NotTo(HaveOccurred())
+
+			Expect(nic.InterfacePropertiesFormat).NotTo(BeNil())
+			Expect(nic.IPConfigurations).NotTo(BeNil())
+			Expect(*nic.IPConfigurations).To(HaveLen(1))
+			Expect((*nic.IPConfigurations)[0].InterfaceIPConfigurationPropertiesFormat).NotTo(BeNil())
+			Expect((*nic.IPConfigurations)[0].PrivateIPAddress).NotTo(BeNil())
+			workerNodes[wn] = *(*nic.IPConfigurations)[0].PrivateIPAddress
+		}
+
+		By("preparing to read resolv.conf from each of the worker nodes by allowing the test namespace's ServiceAccount to use the hostmount-anyuid SecurityContextConstraint")
+		sccName := fmt.Sprintf("system:serviceaccount:%s:default", testNamespace)
+
+		// This is wrapped in an Eventually call with some retries to avoid test flakes on the off chance that
+		// OCP happens to be trying to do stuff to the SecurityContextConstraint at the same time as us.
+		Eventually(func() error {
+			scc, err := clients.SecurityClient.SecurityV1().SecurityContextConstraints().Get(ctx, "hostmount-anyuid", metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+
+			if scc.Users == nil {
+				scc.Users = []string{}
+			}
+			scc.Users = append(scc.Users, sccName)
+			_, err = clients.SecurityClient.SecurityV1().SecurityContextConstraints().Update(ctx, scc, metav1.UpdateOptions{})
+			return err
+		}).WithContext(ctx).
+			WithTimeout(10 * time.Second).
+			WithPolling(1 * time.Second).
+			Should(Succeed())
+
+		DeferCleanup(func(ctx context.Context) {
+			By("removing the test namespace's ServiceAccount's ability to use the hostmount-anyuid SecurityContextConstraint")
+			Eventually(func() error {
+				scc, err := clients.SecurityClient.SecurityV1().SecurityContextConstraints().Get(ctx, "hostmount-anyuid", metav1.GetOptions{})
+				if err != nil {
+					return err
+				}
+
+				if scc.Users == nil {
+					scc.Users = []string{}
+				}
+				idx := -1
+				for i, u := range scc.Users {
+					if u == sccName {
+						idx = i
+						break
+					}
+				}
+				if idx >= 0 {
+					users := []string{}
+					for i, u := range scc.Users {
+						if i == idx {
+							continue
+						}
+						users = append(users, u)
+					}
+					scc.Users = users
+					_, err = clients.SecurityClient.SecurityV1().SecurityContextConstraints().Update(ctx, scc, metav1.UpdateOptions{})
+					return err
+				}
+
+				return nil
+			}).WithContext(ctx).
+				WithTimeout(10 * time.Second).
+				WithPolling(1 * time.Second).
+				Should(Succeed())
+		})
+
+		By("verifying each worker node's resolv.conf via a one-shot Job per node")
+		for wn, ip := range workerNodes {
```

Comment:
I'm wondering if we need to check all three, or if we can save some time by only simulating and checking one node.

Reply:
I thought about this too. I could go either way. I wrote the test to check all three workers because it more accurately simulates an actual occurrence of host servicing in prod, and I don't think it will take _that_ much longer than just doing one worker.

The long bottleneck is the time spent waiting for the nodes to become Ready again after the restart, but it's mostly happening "in parallel" - the actions of starting and stopping don't take long per node.

I would leave it as-is, but I'm open to changing it to just do one worker if you think saving time is a priority.